File: src/main/java/com/example/demo/Entity/BookingDetailTable.java
package com.example.demo.Entity;


import jakarta.persistence.*;
import lombok.Data;


@Entity
@Table(name = "booking_detail_table")
@Data
public class BookingDetailTable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "booking_detail_id")
    private long bookingDetailId;

    @ManyToOne
    @JoinColumn(name = "booking_id")
    private BookingHeaderTable booking;

    @ManyToOne
    @JoinColumn(name = "addon_id")
    private AddOnMaster addon;

    @Column(name = "addon_rate")
    private double addonRate;

}

File: Backend.NET/Models/BookingDetailTable.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FleetManagementSystem.Api.Models;

[Table("booking_detail_table")]
public class BookingDetailTable
{
    [Key]
    [Column("booking_detail_id")]
    public long BookingDetailId { get; set; }

    [ForeignKey("Booking")]
    [Column("booking_id")]
    public long? BookingId { get; set; }
    public BookingHeaderTable Booking { get; set; }

    [ForeignKey("AddOn")]
    [Column("addon_id")]
    public int? AddOnId { get; set; }
    public AddOnMaster AddOn { get; set; }

    [Column("addon_rate")]
    public double? AddonRate { get; set; }
}
